File: sistemaPadronElectoral/sistemaPadronElectoral/Models/Votante.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SistemaPadronElectoral.Models
{
    public class Votante
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Codigo { get; set; }
        public int Ci { get; set; }
        public string Nombre { get; set; }
        public string Apellido { get; set; }
        public string Direccion { get; set; }
        public string Foto { get; set; } = null!;
        public string CiReverso { get; set; } = null!;
        public string CiAnverso { get; set; } = null!;
        public DateOnly FechaNacimiento { get; set; }
        public decimal Latitud { get; set; }

        public decimal Longitud { get; set; } = 0;
        public string Departamento { get; set; } = " ";
        public string Ciudad { get; set; } = " ";
        public string Provincia { get; set; } = " ";
        public int Recinto { get; set; } = 0;
        public Votante()
        {
            Codigo = Guid.NewGuid();
        }
    }
}

File: frontend/src/models/Votante.ts
export interface Votante {
  codigo: number;
  ci: number;
  nombre: string;
  apellido: string;
  direccion: string;
  foto?: File;
  ciReverso?: File;
  ciAnverso?: File;
  fechaNacimiento: string; // Using string for date, can be converted to Date when needed
  latitud: number;
  longitud: number;
  departamento: string;
  ciudad: string;
  provincia: string;
}

// Interface para la consulta pública del padrón (sin datos sensibles)
export interface VotantePublic {
  recinto: string;
  recintoNombre: string;
  codigo: number;
  ci: number;
  nombre: string;
  apellido: string;
  fechaNacimiento: string;
  departamento: string;
  ciudad: string;
  provincia: string;
}

export interface VotanteCreateRequest {
  ci: number;
  nombre: string;
  apellido: string;
  direccion: string;
  foto: File;
  ciReverso: File;
  ciAnverso: File;
  fechaNacimiento: string;
  latitud: number;
  longitud: number;
  departamento: string;
  ciudad: string;
  provincia: string;
  recinto: string; // ID del recinto asociado
}

export interface VotanteUpdateRequest {
  ci: number;
  nombre: string;
  apellido: string;
  direccion: string;
  foto?: File;
  ciReverso?: File;
  ciAnverso?: File;
  fechaNacimiento: string;
  latitud: number;
  longitud: number;
  departamento: string;
  ciudad: string;
  provincia: string;
  recinto: string; // ID del recinto asociado
}
